.travis.yml

```diff
@@ -1,36 +1,45 @@
+sudo: false
 language: python
 
-# Install trollius for 2, 3.2, and pypy
-# Install either trollius or tulip for 3.3
-# Asyncio is in the standard library for 3.4+
 matrix:
-  include:
-    - python: 2.7
-      env: EVENTLOOP=trollius
-    - python: 3.2
-      env: EVENTLOOP=trollius
-    - python: 3.3
-      env: EVENTLOOP=trollius
-    - python: 3.3
-      env: EVENTLOOP=tulip
-    - python: 3.4
-      env: EVENTLOOP=asyncio
-    - python: 3.5
-      env: EVENTLOOP=asyncio
-    - python: pypy
-      env: EVENTLOOP=trollius
-    - python: pypy3
-      env: EVENTLOOP=trollius
-    - python: nightly
-      env: EVENTLOOP=asyncio
-  allow_failures:
-    - python: pypy3
-    - python: nightly
+    include:
+        - python: 2.7  # these are just to make travis's UI a bit prettier
+          env: TOXENV=py27-trollius
+        - python: 3.2
+          env: TOXENV=py32-trollius
+        - python: 3.3
+          env: TOXENV=py33-trollius
+        - python: 3.3
+          env: TOXENV=py33-tulip
+        - python: 3.4
+          env: TOXENV=py34
+        - python: 3.5
+          env: TOXENV=py35
+        - python: nightly
+          env: TOXENV=py-nightly
 
-script:
-  - make lint
-  - make check-cov
+        - python: pypy
+          env: TOXENV=pypy-trollius
+        - python: pypy3
+          env: TOXENV=pypy3-trollius
 
+        - python: 3.5
+          env: TOXENV=packaging
+        - python: 3.5
+          env: TOXENV=docs
+        - python: 3.5
+          env: TOXENV=py3-pep8
+        - python: 2.7
+          env: TOXENV=py2-pep8
+
+    allow_failures:
+        - python: nightly
+          env: TOXENV=py-nightly
+        - python: pypy
```

Comment:
Yeah, I guess we need this for now, but it is a very consistent bug, and seemed to always be in the same test, but it's hard to say if the bug is in qtile, the tests, or from running on 12.04.